pkg/apiserver/rest/dualwriter.go

```diff
@@ -4,32 +4,33 @@ import (
 	"context"
 	"errors"
 
-	"k8s.io/apimachinery/pkg/api/meta"
-	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apiserver/pkg/registry/rest"
-	"k8s.io/klog"
 )
 
 var (
-	_ rest.Storage              = (*DualWriter)(nil)
-	_ rest.Scoper               = (*DualWriter)(nil)
-	_ rest.TableConvertor       = (*DualWriter)(nil)
-	_ rest.CreaterUpdater       = (*DualWriter)(nil)
-	_ rest.CollectionDeleter    = (*DualWriter)(nil)
-	_ rest.GracefulDeleter      = (*DualWriter)(nil)
-	_ rest.SingularNameProvider = (*DualWriter)(nil)
+	_ rest.Storage              = (DualWriter)(nil)
```

Comment:
i think these checks can be removed since they were originally checking to make sure a struct implemented all of the expected interfaces

Reply:
I'd rather keep them for now since we will have to maintain 4 implementations of the dual writer: they can serve as guidelines so that we don't forget all the methods we need to implement. Specially since we don't have concrete implementations for all the methods yet